
  Error: not folder path
      at C:\Users\lithd\Documents\GitHub\ennv\site\server\build\request\file.js:152:15
      at dispatch (C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\koa-compose@4.1.0\node_modules\koa-compose\index.js:42:32)
      at C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\koa-router@12.0.0\node_modules\koa-router\lib\router.js:425:16
      at dispatch (C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\koa-compose@4.1.0\node_modules\koa-compose\index.js:42:32)
      at C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\koa-compose@4.1.0\node_modules\koa-compose\index.js:34:12
      at dispatch (C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\koa-router@12.0.0\node_modules\koa-router\lib\router.js:430:31)
      at dispatch (C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\koa-compose@4.1.0\node_modules\koa-compose\index.js:42:32)
      at C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\koa-mount@4.0.0\node_modules\koa-mount\index.js:52:32
      at dispatch (C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\koa-compose@4.1.0\node_modules\koa-compose\index.js:42:32)
      at bodyParser (C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\koa-bodyparser@4.4.0\node_modules\koa-bodyparser\index.js:91:11)

TypeError: Middleware stack must be an array!
    at compose (C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\koa-compose@4.1.0\node_modules\koa-compose\index.js:20:41)
    at Server.callback (C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\koa@2.14.2\node_modules\koa\lib\application.js:150:16)
    at Server.emit (node:events:513:28)
    at Server.<anonymous> (C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\@pm2+io@5.0.0\node_modules\@pm2\io\build\main\metrics\httpMetrics.js:166:33)
    at parserOnIncoming (node:_http_server:1091:12)
    at HTTPParser.parserOnHeadersComplete (node:_http_common:119:17)
TypeError: Middleware stack must be an array!
    at compose (C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\koa-compose@4.1.0\node_modules\koa-compose\index.js:20:41)
    at Server.callback (C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\koa@2.14.2\node_modules\koa\lib\application.js:150:16)
    at Server.emit (node:events:513:28)
    at Server.<anonymous> (C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\@pm2+io@5.0.0\node_modules\@pm2\io\build\main\metrics\httpMetrics.js:166:33)
    at parserOnIncoming (node:_http_server:1091:12)
    at HTTPParser.parserOnHeadersComplete (node:_http_common:119:17)
TypeError [ERR_INVALID_URL]: Invalid URL
    at new NodeError (node:internal/errors:399:5)
    at new URL (node:internal/url:560:13)
    at WebSocketServer.<anonymous> (C:\Users\lithd\Documents\GitHub\ennv\site\server\build\request\connect.js:73:22)
    at WebSocketServer.emit (node:events:513:28)
    at WebSocketServer.completeUpgrade (C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\ws@8.13.0\node_modules\ws\lib\websocket-server.js:431:5)
    at WebSocketServer.handleUpgrade (C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\ws@8.13.0\node_modules\ws\lib\websocket-server.js:339:10)
    at Server.upgrade (C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\ws@8.13.0\node_modules\ws\lib\websocket-server.js:114:16)
    at Server.emit (node:events:513:28)
    at Server.<anonymous> (C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\@pm2+io@5.0.0\node_modules\@pm2\io\build\main\metrics\httpMetrics.js:152:37)
    at onParserExecuteCommon (node:_http_server:903:14) {
  input: '/connect',
  code: 'ERR_INVALID_URL'
}
TypeError [ERR_INVALID_URL]: Invalid URL
    at new NodeError (node:internal/errors:399:5)
    at new URL (node:internal/url:560:13)
    at WebSocketServer.<anonymous> (C:\Users\lithd\Documents\GitHub\ennv\site\server\build\request\connect.js:73:22)
    at WebSocketServer.emit (node:events:513:28)
    at WebSocketServer.completeUpgrade (C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\ws@8.13.0\node_modules\ws\lib\websocket-server.js:431:5)
    at WebSocketServer.handleUpgrade (C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\ws@8.13.0\node_modules\ws\lib\websocket-server.js:339:10)
    at Server.upgrade (C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\ws@8.13.0\node_modules\ws\lib\websocket-server.js:114:16)
    at Server.emit (node:events:513:28)
    at Server.<anonymous> (C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\@pm2+io@5.0.0\node_modules\@pm2\io\build\main\metrics\httpMetrics.js:152:37)
    at onParserExecuteCommon (node:_http_server:903:14) {
  input: '/connect',
  code: 'ERR_INVALID_URL'
}
TypeError [ERR_INVALID_URL]: Invalid URL
    at new NodeError (node:internal/errors:399:5)
    at new URL (node:internal/url:560:13)
    at WebSocketServer.<anonymous> (C:\Users\lithd\Documents\GitHub\ennv\site\server\build\request\connect.js:73:22)
    at WebSocketServer.emit (node:events:513:28)
    at WebSocketServer.completeUpgrade (C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\ws@8.13.0\node_modules\ws\lib\websocket-server.js:431:5)
    at WebSocketServer.handleUpgrade (C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\ws@8.13.0\node_modules\ws\lib\websocket-server.js:339:10)
    at Server.upgrade (C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\ws@8.13.0\node_modules\ws\lib\websocket-server.js:114:16)
    at Server.emit (node:events:513:28)
    at Server.<anonymous> (C:\Users\lithd\Documents\GitHub\ennv\node_modules\.pnpm\@pm2+io@5.0.0\node_modules\@pm2\io\build\main\metrics\httpMetrics.js:152:37)
    at onParserExecuteCommon (node:_http_server:903:14) {
  input: '/message/connect/',
  code: 'ERR_INVALID_URL'
}
